### PR TITLE
Clear result box

### LINKED ISSUE #140: [PE-D][Tester D] Date values with invalid dates can be accepted
It seems like somehow, your date parser is able to accept date values greater than 30 or 31 (as we know a month can have 28,29,30,31 days depending on which month it is in the year). Perhaps you could use a built-in java `LocalDate` parser which would flag out these invalid dates.

It appears that the same thing could happen for months also (seems like the difference between the month value input and what is accepted (should be less than or equals to 31), is being translated into years.

You might want to investigate how your command parser is handling the parsing of dates from the string in the expected format.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: timleow/ped#3

### LINKED ISSUE #148: [PE-D][Tester C] View: Feature allows additional input after a whitespace
Putting extra input after a `view o`/`view s` command is still treated as a valid input

input: `view o hello`

output: 

Maybe consider classifying such situations as invalid instead

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: phiphi-tan/ped#8

### LINKED ISSUE #172: [PE-D][Tester C] UG: Mismatching description for Clear
UG describes the `clear` feature as "Clears the output of the previous command.", but in-app it clears the entire result box, which means it clears all command history displayed. Might want to update either the feature or the UG to match its actual functionality/description.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: phiphi-tan/ped#17

### PR DESCRIPTION
**Pull request by AlainS87**
Fixes #140 
Fixes #172 
Fixes #148

### CODE REVIEW

**Review by cxo05 — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md [hunk 1]
```diff
@@ -318,13 +319,13 @@ Examples:
 - `find John` returns `john` and `John Doe`
 - `find alex david` returns `Alex Yeoh`, `David Li`
 
-### Clearing the output : `clear`
+### Clearing the output : `clearresultbox`
```
[the remark this answers is not in the document]
Comment (AlainS87) [reply docs/UserGuide.md]: renamed back

[the remark this answers is not in the document]
Comment (cxo05) [reply docs/UserGuide.md]: Missed out line 143 in the UG

[the remark this answers is not in the document]
Comment (AlainS87) [reply docs/UserGuide.md]: sry i was abit careless just now

**Review by cxo05 — APPROVED** [review]
LGTM

**Pull request merged**